plugin/kubernetes/kubernetes.go

```diff
@@ -276,10 +279,13 @@ func (k *Kubernetes) Records(state request.Request, exact bool) ([]msg.Service,
 	return services, err
 }
 
-func endpointHostname(addr api.EndpointAddress) string {
+func endpointHostname(addr api.EndpointAddress, endpointNameMode string) string {
 	if addr.Hostname != "" {
 		return strings.ToLower(addr.Hostname)
 	}
+	if endpointNameMode == endpointNameModeName && addr.TargetRef != nil && addr.TargetRef.Name != "" {
```

Comment:
Actually now that I think some more; endpointNameMode is just a boolean, no need to carry a string with us all time and to a string compare - this is either on or off.